src/NamedPipe.re

```diff
@@ -1,10 +1,19 @@
 type t = string;
 
+module Internal = {
+  let id = ref(0);
```

Comment:
This is not an id, it's a reference to the last id generated, it's also poorly encapsulated. Why should it be possible to use `Internal.id` anywhere in this module?

Reply:
I moved it to be encapsulated in the `create` method (and renamed from `id` to `lastId`) 👍